lib/puppet/functions/stdlib/nested_values.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# @summary Get list of nested values from given hash
+#   This function will return list of nested Hash values and returns list of values in form of Array
+#
+# @example Example Usage:
+#   $hash = {
+#     "key1" => "value1",
+#     "key2" => { "key2.1" => "value2.1"},
+#     "key3" => "value3"
+#   }
+#   $data = $hash.stdlib::nested_values
+#   #Output : ["value1", "value2.1", "value3"]
+Puppet::Functions.create_function(:'stdlib::nested_values') do
+  # @param options A (nested) hash
+  # @return [Data]
```

Comment:
`Data`? Shouldn't this be `Array`? (and I wonder if it's needed given the `return_type` is also specified in the dispatch.

```suggestion
  # @return All the values found in the input hash included those deeply nested.
```

I'm also wondering if this could/should be expanded to handle data in _arrays_ inside in the hash??